src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -252,7 +252,7 @@ public Future<Optional<InputStream>> getRecording(
     }
 
     public Future<Void> deleteRecording(
-            ConnectionDescriptor connectionDescriptor, String recordingName) {
+            ConnectionDescriptor connectionDescriptor, String recordingName, boolean isAnUnreadableSnapshot) {
```

Comment:
Can/would callers from outside of this class be expected to know whether what they're deleting is an unreadable snapshot? The two examples here in the DeleteActiveRecordingMutator and the TargetRecordingDeleteHandler both pass `false` here, and I think that's likely to always be the case for other mutators/handlers.

I think it might be "nicer" to have methods like this:

```java
public Future<Void> deleteRecording(ConnectionDescriptor connectionDescriptor, String recordingName);
Future<Void> deleteRecording(ConnectionDescriptor connectionDescriptor, String recordingName, boolean sendNotification);
```

The mutators/handlers would call the `public` one, which in turn just wraps around the package-private one (which might also be better suited as a `protected` or `private`) and calls it with `true` for the final parameter, which is an inverted condition compared to how it is now.